sign/eddilithium3/example_test.go

```diff
@@ -0,0 +1,45 @@
+package eddilithium3_test
+
+import (
+	"fmt"
+
+	"github.com/cloudflare/circl/sign/eddilithium3"
+)
+
+func Example() {
+	// Generates a keypair.
+	pk, sk, err := eddilithium3.GenerateKey(nil)
```

Comment:
```suggestion
	pk, sk, err := eddilithium3.GenerateKey(rand.Reader)
```

Reply:
Specifying the random source explicitly as `crypto/rand` can only lead to issues if `rand` happens to point to something else than `crypto/rand` and doesn't have any advantage as `crypto/rand` is the default anyway.  What am I missing?

Reply:
This is not about functionality, it's about best practices and educating people. 
Even though, the internals of `GenerateKey` replace nil by `crypto/rand`, putting nil in the example, gives a bad example.

Reply:
Why is it bad?  I think `rand.Reader` is strictly worse than `nil` because `rand` could be something else than `crypto/rand`.

Reply:
I agree with @armfazh. Passing `crypto/rand` in is best practice, and we ought to make the test code reflect best practice.

Reply:
Could you please explain *why* this is bad practice.

Reply:
Some input https://checkmarx.gitbooks.io/go-scp/cryptographic-practices/pseudo-random-generators.html

Reply:
(already discussed internally)